pkg/operator/prereqs_v311_00.go

```diff
@@ -0,0 +1,145 @@
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v311_00_assets"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1alpha1helpers"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// createPrereqs_v311_00_to_latest takes care of creation of valid resources in a fixed name.  These are inputs to other control loops.
+// returns whether or not requeue and if an error happened when updating status.  Normally it updates status itself.
+func createPrereqs_v311_00_to_latest(c Prereqs, operatorConfig *v1alpha1.KubeApiserverOperatorConfig) (bool, error) {
+	operatorConfigOriginal := operatorConfig.DeepCopy()
+	errors := []error{}
+
+	directResourceResults := resourceapply.ApplyDirectly(c.kubeClient, v311_00_assets.Asset,
+		"v3.11.0/kube-apiserver/ns.yaml",
+		"v3.11.0/kube-apiserver/public-info-role.yaml",
+		"v3.11.0/kube-apiserver/public-info-rolebinding.yaml",
+		"v3.11.0/kube-apiserver/svc.yaml",
+	)
+
+	for _, currResult := range directResourceResults {
+		if currResult.Error != nil {
+			errors = append(errors, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+		}
+	}
+
+	apiserverConfig, _, err := manageKubeApiserverConfigMap_v311_00_to_latest(c.kubeClient.CoreV1(), operatorConfig)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "configmap/deployment-kube-apiserver-config", err))
+	}
+	_, _, err = managePod_v311_00_to_latest(c.kubeClient.CoreV1(), operatorConfig)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "configmap/kube-apiserver-pod", err))
+	}
+
+	configData := ""
+	if apiserverConfig != nil {
+		configData = apiserverConfig.Data["config.yaml"]
+	}
+	_, _, err = manageKubeApiserverPublicConfigMap_v311_00_to_latest(c.kubeClient.CoreV1(), configData, operatorConfig)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q: %v", "configmap/public-info", err))
+	}
+
+	if len(errors) > 0 {
+		message := ""
+		for _, err := range errors {
+			message = message + err.Error() + "\n"
+		}
+		v1alpha1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1alpha1.OperatorCondition{
+			Type:    "PrereqCreationFailing",
+			Status:  operatorv1alpha1.ConditionTrue,
+			Reason:  "SynchronizationError",
+			Message: message,
+		})
+		if !reflect.DeepEqual(operatorConfigOriginal, operatorConfig) {
+			_, updateError := c.operatorConfigClient.KubeApiserverOperatorConfigs().UpdateStatus(operatorConfig)
+			return true, updateError
+		}
+		return true, nil
+	}
+
+	v1alpha1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1alpha1.OperatorCondition{
+		Type:   "PrereqCreationFailing",
```

Comment:
"prereq" is confusing as a name. See other comment. "TargetConfigCreationFailed" would be much clearer.